microservices-connector/cmd/router/main.go

```diff
@@ -365,22 +367,55 @@ func routeStep(nodeName string, graph mcv1alpha3.GMConnector, input []byte, head
 }
 
 func mcGraphHandler(w http.ResponseWriter, req *http.Request) {
-	inputBytes, _ := io.ReadAll(req.Body)
-	if response, statusCode, err := routeStep(defaultNodeName, *mcGraph, inputBytes, req.Header); err != nil {
-		log.Error(err, "failed to process request")
-		w.Header().Set("Content-Type", "application/json")
-		w.WriteHeader(statusCode)
-		if _, err := w.Write(prepareErrorResponse(err, "Failed to process request")); err != nil {
-			log.Error(err, "failed to write mcGraphHandler response")
+	ctx, cancel := context.WithTimeout(req.Context(), time.Minute)
+	defer cancel()
+
+	done := make(chan struct{})
+	go func() {
+		defer close(done)
+
+		inputBytes, _ := io.ReadAll(req.Body)
+		response, statusCode, err := routeStep(defaultNodeName, *mcGraph, inputBytes, req.Header)
+
+		if err != nil {
+			log.Error(err, "failed to process request")
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(statusCode)
+			if _, err := w.Write(prepareErrorResponse(err, "Failed to process request")); err != nil {
+				log.Error(err, "failed to write mcGraphHandler response")
+			}
+			return
 		}
-	} else {
 		if json.Valid(response) {
 			w.Header().Set("Content-Type", "application/json")
```

Comment:
if json invalid what do we set in content-type? blank/nothing? is that valid?

Reply:
There seem to be several types looking at https://stackoverflow.com/questions/23714383/what-are-all-the-possible-values-for-http-content-type-header  and we might encounter these when using genAI for image, audio etc construction. Perhaps add a note "Other types not validated at this time" type comment.

Reply:
Hi @mkbhanda, it's a great question, router service as an entry endpoint, both the requests and responses of microservices are based on json format, so it's oaky for now. And I have some tests that the image can also be represented as base64 encode data.
Of course, there would be other changes when some other examples imported. Thanks for raising this question!